sphinxcontrib/devhelp/__init__.py

```diff
@@ -109,37 +113,24 @@ def write_toc(node, parent):
         for node in tocdoc.traverse(matcher):  # type: addnodes.compact_paragraph
             write_toc(node, chapters)
 
-        # Index
+        # Index keywords
         functions = etree.SubElement(root, 'functions')
-        index = IndexEntries(self.env).create_index(self)
-
-        def write_index(title, refs, subitems):
-            # type: (str, List[Any], Any) -> None
-            if len(refs) == 0:
-                pass
-            elif len(refs) == 1:
-                etree.SubElement(functions, 'function',
-                                 name=title, link=refs[0][1])
-            else:
-                for i, ref in enumerate(refs):
-                    etree.SubElement(functions, 'function',
-                                     name="[%d] %s" % (i, title),
-                                     link=ref[1])
-
-            if subitems:
-                parent_title = re.sub(r'\s*\(.*\)\s*$', '', title)
-                for subitem in subitems:
-                    write_index("%s %s" % (parent_title, subitem[0]),
-                                subitem[1], [])
-
-        for (key, group) in index:
-            for title, (refs, subitems, key) in group:
-                write_index(title, refs, subitems)
+        if self.domain_indices:
+            for domain_name in sorted(self.env.domains):
+                for domain_data in self.env.domains[domain_name].get_objects():
+                    name, _, _type, docname, anchor, _ = domain_data
+                    link = '{docname}.html#{anchor}'.format(docname=docname,
+                                                            anchor=anchor)
+                    etree.SubElement(functions, 'keyword',
+                                     name=name, link=link, type=_type)
 
         # Dump the XML file
-        xmlfile = path.join(outdir, outname + '.devhelp.gz')
-        with gzip.open(xmlfile, 'w') as f:
-            tree.write(f, 'utf-8')
+        xmlfile = path.join(outdir, outname + '.devhelp2')
+        with codecs.open(xmlfile, 'a', 'utf-8') as f:
```

Comment:
Since python3, `open()` function supports to change encodings. There are no reason to use `codecs.open()` for that purpose now.